src/lib/filelock/filelock_windows.go

```diff
@@ -0,0 +1,117 @@
+package filelock
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	kernel32     = windows.NewLazySystemDLL("kernel32.dll")
+	lockFileEx   = kernel32.NewProc("LockFileEx")
+	unlockFileEx = kernel32.NewProc("UnlockFileEx")
+)
+
+// Open will open and lock a file.  It blocks until the lock is acquired.
+// If the file does not yet exist, it creates the file, and any missing
+// directories above it in the path.  To release the lock, Close the file.
+func (l *locker) Open() (LockedFile, error) {
+	dir := filepath.Dir(l.path)
+	err := os.MkdirAll(dir, 0700)
+	if err != nil {
+		panic(err)
```

Comment:
@rosenhouse actually, its in the original code: https://github.com/cloudfoundry-incubator/cf-networking-release/blob/develop/src/lib/filelock/filelock.go#L25

should we remove that as part of this PR?